File: sponge-standalone/examples/standalone/standalone_camel_example/standalone_camel_example.rb
# Sponge Knowledge base
# Standalone with Camel example

java_import org.apache.camel.builder.RouteBuilder

class RubyRoute < RouteBuilder
    def configure
        self.from("sponge:spongeEngine").routeId("spongeConsumerCamelRuby") \
                .transform().simple("Ruby route - Received message: ${body}") \
                .to("stream:out")
    end
end

def onStartup
    $camel.context.addRoutes(RubyRoute.new())
end

File: sponge-standalone/examples/standalone/standalone_camel_example/standalone_camel_example.py
"""
Sponge Knowledge base
Standalone with Camel example
"""

from org.openksavi.sponge.camel import ScriptRouteBuilder

def onInit():
    # Variables for assertions only
    EPS.setVariable("sent", False)
    EPS.setVariable("message", None)

class CamelTrigger(Trigger):
    def configure(self):
        self.event = "spongeEvent"
    def run(self, event):
        message = spring.context.getBean("spongeEngine").description + ": " + event.get("message")
        camel.send(message)
        EPS.setVariable("sent", True)

class PythonRoute(ScriptRouteBuilder):
    def configure(self):
        self.fromS("sponge:spongeEngine").routeId("spongeConsumerCamelPython") \
                .transform().simple("Python route - Received message: ${body}") \
                .process(lambda exchange: EPS.setVariable("message", exchange.getIn().getBody())) \
                .to("stream:out")

def onStartup():
    camel.context.addRoutes(PythonRoute())
    EPS.event("spongeEvent").set("message", "Send me to Camel").send()
